pkg/image/api/helper.go

```diff
@@ -610,6 +634,34 @@ func ResolveLatestTaggedImage(stream *ImageStream, tag string) (string, bool) {
 	}
 }
 
+// DockerImageReferenceForImage returns the docker reference for specified image. Assuming
+// the image stream contain the image and the image has corresponding tag, this function
+// will try to find this tag and take the reference policy into the account.
+// The the image stream does not reference the image or the image does not have
+// corresponding tag event, this function will return false.
+func DockerImageReferenceForImage(stream *ImageStream, imageID string) (string, bool) {
+	tag, event := LatestImageTagEvent(stream, imageID)
+	if len(tag) == 0 {
+		return "", false
+	}
+	ref, ok := stream.Spec.Tags[tag]
+	if !ok {
+		return event.DockerImageReference, true
+	}
+	switch ref.ReferencePolicy.Type {
+	case LocalTagReferencePolicy:
+		ref, err := ParseDockerImageReference(event.DockerImageReference)
```

Comment:
Use `stream.Status.DockerImageRepository` as a base reference. Override its ID and unset its Tag.